apps/judicial-system/web/src/routes/Shared/AppealToCourtOfAppeals/AppealCaseFiles.tsx

```diff
@@ -156,16 +195,20 @@ const AppealFiles = () => {
       <FormContentContainer isFooter>
         <FormFooter
           previousUrl={previousUrl}
-          onNextButtonClick={async () => {
-            sendNotification(
-              workingCase.id,
-              NotificationType.APPEAL_CASE_FILES_UPDATED,
-            )
-
-            setVisibleModal(true)
-          }}
-          nextButtonText={formatMessage(strings.nextButtonText)}
+          onNextButtonClick={handleNextButtonClick}
+          nextButtonText={formatMessage(
+            uploadState.error
+              ? strings.uploadFailedNextButtonText
+              : strings.nextButtonText,
+          )}
           nextButtonIcon={undefined}
+          nextIsLoading={uploadState.isUploading}
+          nextIsDisabled={newFiles.length === 0}
+          nextButtonColorScheme={
+            uploadState.error && !uploadState.isUploading && newFiles.length > 0
+              ? 'destructive'
+              : 'default'
+          }
```

Comment:
I tried this locally and only a single file is uploaded even if I select multiple files. Not sure what is happening.